apis/python/node/src/lib.rs

```diff
@@ -218,9 +218,15 @@ impl Node {
                         value
                             .to_pyarrow(py)
                             .context("failed to convert value to pyarrow")
-                            .unwrap_or_else(|err| PyErr::from(err).to_object(py))
+                            .unwrap_or_else(|err| match PyErr::from(err).into_pyobject(py) {
+                                Ok(py_object) => py_object.unbind().into(),
+                                Err(_) => py.None(),
```

Comment:
I don't think that we should silently return None in case of error here. Can you use `.context("Could not return error")?`.

Resulting in .None() here would make it very hard to debug.